syntax = "proto3";

import "google/protobuf/timestamp.proto";

service EventService {
  rpc get_event_by_event_id(EventRequestByEventId) returns (EventResponse){}
  rpc get_events_by_events_ids(EventsRequestByEventsIds) returns (EventsResponse){}
  rpc get_events_by_author_id(EventsRequestByAuthorId) returns (EventsResponse){}
  rpc create_event(Event) returns (BaseResponse){}
  rpc update_event(Event) returns (BaseResponse){}
  rpc delete_event(DeleteEventRequest) returns (BaseResponse){}
}

message EventsRequestByAuthorId {
  string author_id = 1;
}

message EventRequestByEventId {
  string event_id = 1;
}

message EventsRequestByEventsIds {
  ListOfEventsIds events_ids = 1;
}

message EventResponse {
  int32 status_code = 1;
  Event event = 2;
  optional string message = 3;
}

message EventsResponse {
  int32 status_code = 1;
  optional ListOfEvents events = 2;
  optional string message = 3;
}

message Event {
  string id = 1;
  string title = 2;
  optional string description = 3;
  string color = 4;
  google.protobuf.Timestamp start = 5;
  google.protobuf.Timestamp end = 6;
  optional google.protobuf.Timestamp repeating_delay = 7;
  string author_id = 8;
}

message ListOfEvents {
  repeated Event events = 1;
}

message ListOfEventsIds {
  repeated string ids = 1;
}

message BaseResponse {
  int64 status_code = 1;
  optional string message = 2;
}

message DeleteEventRequest {
  string event_id = 1;
}
